works around things like

 third_party/zlib/google/zip_reader.cc:99:24: error: variable has incomplete type 'base::stat_wrapper_t' (aka 'stat64')
   base::stat_wrapper_t sb;
                       ^
 base/files/file.h:28:8: note: forward declaration of 'stat64'
 struct stat64;

should be resolved by https://chromium-review.googlesource.com/c/chromium/src/+/4134763

--- a/base/files/file.h
+++ b/base/files/file.h
@@ -19,7 +19,8 @@
 #include "build/build_config.h"
 
 #if BUILDFLAG(IS_BSD) || BUILDFLAG(IS_APPLE) || BUILDFLAG(IS_NACL) || \
-    BUILDFLAG(IS_FUCHSIA) || (BUILDFLAG(IS_ANDROID) && __ANDROID_API__ < 21)
+    BUILDFLAG(IS_FUCHSIA) || (BUILDFLAG(IS_ANDROID) && __ANDROID_API__ < 21) || \
+    (defined(OS_LINUX) && !defined(__GLIBC__))
 struct stat;
 namespace base {
 typedef struct stat stat_wrapper_t;
